### PR TITLE
Cleaning up subdomains management

### PR DESCRIPTION
**Pull request by DenysGonchar**
this PR introduces the following changes:
* all subdomains are registered using `mongoose_domain_api` interfaces - this change is required for mod_domain_isolation module.
* removal of mongoose_subhosts
* fix of mod_muc_iq generic iq component

### CODE REVIEW

File: src/mod_muc_room.erl [hunk 1]
```diff
@@ -4581,12 +4581,12 @@ route_iq(Acc, #routed_iq{iq = #iq{type = Type, xmlns = ?NS_DISCO_ITEMS, lang = L
...
 route_iq(Acc, #routed_iq{iq = IQ = #iq{}, packet = Packet, from = From},
-         #state{host_type = HostType, jid = RoomJID} = StateData) ->
+         #state{host = Host, jid = RoomJID} = StateData) ->
     %% Custom IQ, addressed to this room's JID.
-    case mod_muc_iq:process_iq(HostType, From, RoomJID, Acc, IQ) of
+    case mod_muc_iq:process_iq(Host, From, RoomJID, Acc, IQ) of
         {Acc1, error} ->
             ?LOG_WARNING(#{what => muc_process_iq_failed, acc => Acc,
-                           host_type => HostType, room_jid => RoomJID}),
+                           host => Host, room_jid => RoomJID}),
```
Comment (arcusfelis) [inline src/mod_muc_room.erl:4589]: log host_type field too.
host field is called `server`

File: src/mod_muc_iq.erl [hunk 2]
```diff
@@ -36,24 +36,24 @@ start_link() ->
 
 %% @doc Handle custom IQ.
 %% Called from mod_muc_room.
--spec process_iq(mongooseim:host_type(), jid:jid(), jid:jid(), mongoose_acc:t(),
+-spec process_iq(jid:server(), jid:jid(), jid:jid(), mongoose_acc:t(),
         jlib:iq()) -> mongoose_acc:t() | {mongoose_acc:t(), error}.
-process_iq(HostType, From, RoomJID, Acc, IQ = #iq{xmlns = XMLNS}) ->
-    case ets:lookup(tbl_name(), {XMLNS, HostType}) of
+process_iq(Host, From, RoomJID, Acc, IQ = #iq{xmlns = XMLNS}) ->
```
Comment (arcusfelis) [inline src/mod_muc_iq.erl:41]: can we name it MucHost?
also type is probably jid:lserver()

**Review by arcusfelis — COMMENTED** [review]
added some comments

File: src/ejabberd_router.erl [hunk 3]
```diff
@@ -268,9 +277,9 @@ check_component_global(LDomain, Node) ->
     %% check that there is no component registered globally for this node
     case get_global_component(LDomain, Node) of
         undefined ->
-            ok;
+            false;
         _ ->
-            false
+            true
```
Comment (chrzaszcz) [inline src/ejabberd_router.erl:282]: Ok, so we had a hidden bug, maybe we need to add a test?

**Review by chrzaszcz — APPROVED** [review]
Looks good, I have one minor comment.

File: src/ejabberd_router.erl (hunk 3 above)
Comment (DenysGonchar) [reply src/ejabberd_router.erl:282]: Not a bug, it was working correctly. But I think that having a validation function that returns `boolean() | ok`, when only `ok`value indicates a successful check, is super confusing. So I unified and reworked it a bit.

**Pull request merged**